lib/galaxy/managers/jobs.py

```diff
@@ -84,7 +84,7 @@ def by_tool_input(self, trans, tool_id, tool_version, param=None, param_dump=Non
         def populate_input_data_input_id(path, key, value):
             """Traverses expanded incoming using remap and collects input_ids and input_data."""
             if key == 'id':
-                path_key = get_path_key(path[:-2])
+                path_key = get_path_key(path[:-2]).split('|')[-1]
```

Comment:
That's some serious detective work here! Interesting that this fixes things for you, that is probably only correct for a specific parameter type or combination.
I'm worried about matching the wrong input, especially since `get_path_key` goes to great lengths trying to maintain the `|` separated form.
Could you start by adding a test case that fails without this change ?  There is one test in https://github.com/galaxyproject/galaxy/blob/release_20.01/lib/galaxy_test/api/test_tools.py#L630, and now the challenge would be to figure out which dataset parameter is stored without the prefix. Once you know that you can either use an existing framework tool (in test/functional/tools) that fails in the same way or construct a new one for this purpose.

Reply:
Thank you! Let me have a good at creating the test and seeing if I can find which parameter we're looking for.